diff --git a/node_modules/ipfs-bitswap/src/network.js b/node_modules/ipfs-bitswap/src/network.js
index 2b64672..65e15cd 100644
--- a/node_modules/ipfs-bitswap/src/network.js
+++ b/node_modules/ipfs-bitswap/src/network.js
@@ -84,7 +84,7 @@ export class Network {
     this._registrarIds = []
 
     for (const protocol of this._protocols) {
-      this._registrarIds.push(await this._libp2p.register(protocol, topology))
+      this._registrarIds.push(await this._libp2p.registrar.register(protocol, topology))
     }
 
     // All existing connections are like new ones for us
@@ -102,7 +102,7 @@ export class Network {
     // unregister protocol and handlers
     if (this._registrarIds != null) {
       for (const id of this._registrarIds) {
-        this._libp2p.unregister(id)
+        this._libp2p.registrar.unregister(id)
       }
 
       this._registrarIds = []
