diff --git a/node_modules/libp2p/dist/src/libp2p.js b/node_modules/libp2p/dist/src/libp2p.js
index 7a98456..4e365a1 100644
--- a/node_modules/libp2p/dist/src/libp2p.js
+++ b/node_modules/libp2p/dist/src/libp2p.js
@@ -270,12 +270,15 @@ export class Libp2pNode extends EventEmitter {
         if (this.keychain == null) {
             return;
         }
-        try {
-            await this.keychain.findKeyByName('self');
-        }
-        catch (err) {
-            await this.keychain.importPeer('self', this.peerId);
+        
+        const keyInfos = await this.keychain.listKeys()
+        for (const keyInfo of keyInfos){
+            if (keyInfo.name === 'self'){
+                await this.keychain.findKeyByName('self')
+                return
+            }
         }
+        await this.keychain.importPeer('self', this.peerId)
     }
     isStarted() {
         return this.started;
